### ISSUE TITLE
Allow disabling git usage

### ISSUE THREAD

**Issue by CosmicToast** [xwmx/nb#97]
I'm handling nb syncing by way of [syncthing](https://syncthing.net/) and have no need/interest in the history features.
It would be nice if I could save on the `.git` filetree when it comes to syncing and just sync the data itself.

I'm not sure whether or not `nb` uses any git features beyond for those two use-cases though, so this may or may not be easily feasible.
Obviously, no-git would imply no-sync (until perhaps other sync methods are included, rsync perhaps?).

This isn't really a huge priority, but I think it would be nice (especially for syncing with mobile, where the `.git` directory might not always be hidden for plain editors).

**Comment by xwmx** [comment]
@CosmicToast Indeed, that makes sense, and I’m glad you brought this up. `nb search` also currently relies on `git grep`. Specifically the `AND` query implementation needs to be generalized so it works with other search tools. There is already a `$_GIT_ENABLED` global variable for toggling git in some areas, and that’ll need to be added to all the git code paths to make everything still works nicely with it off. I’ll work toward this.

**Comment by maxdiebold-erg** [comment]
To add to this, with #198 in mind, I'm not a fan of the need to set the user and email globally for `nb` to function. I think, before disabling git entirely, since there are already options to toggle off sync (`NB_AUTO_SYNC`) and set an individual notebook's author (`nb notebooks author <notebook-name>`), it may be simpler to prompt users for the author settings of each notebook if they're not already set and there is no global git user config.

I use [git-sync](https://github.com/simonthum/git-sync) to sync my notebooks, so having each notebook in a repository is useful for me, but the global git config requirement hampers a lot of other activities.

**Comment by dukeofcool199** [comment]
is this feature still being looked into?  I would really like to manage my nb notebooks in [git-annex](https://git-annex.branchable.com/), and being able to turn off git would make that integration significantly easier.

**Comment by dukeofcool199** [comment]
i actually found a pretty neat emergent behaviour.

I can make a folder in my git-annex repo the nb_dir, and setup git-annex to make git-add always add stuff to the annex instead of to git.  This way i get the best of both worlds with git-annex being able to store my files, and allow nb to work its autosaving git-magic.

the only weird quirk I have really had to workaround so far is that if I want to make a new notebook, I have to go into the nb_dir, and make one manually and add a file myself so that nb doesn't try to make that folder a git repo by default.
